openedx/core/djangoapps/content/learning_sequences/migrations/0001_initial.py

```diff
@@ -3,122 +3,159 @@
-        ),
-        migrations.AddField(
-            model_name='coursesection',
-            name='learning_context',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='sections', to='learning_sequences.LearningContext'),
-        ),
-        migrations.AlterUniqueTogether(
-            name='learningsequence',
-            unique_together={('learning_context', 'usage_key')},
-        ),
-        migrations.AlterUniqueTogether(
-            name='coursesectionsequence',
-            unique_together={('learning_context', 'ordering')},
-        ),
-        migrations.AlterUniqueTogether(
-            name='coursesection',
-            unique_together={('learning_context', 'usage_key')},
-        ),
-        migrations.AlterIndexTogether(
-            name='coursesection',
-            index_together={('learning_context', 'ordering')},
-        ),
+def run_before_migrate(migrations, db_engine):
+    if 'postgresql' in db_engine:
+        # PostgreSQL: Use binary collation
+        return [
+            migrations.RunSQL(
+                'ALTER TABLE learning_sequences_learningcontext ALTER COLUMN context_key TYPE VARCHAR(255) COLLATE "C";',
+                reverse_sql=migrations.RunSQL.noop, ),
+            migrations.RunSQL(
+                'ALTER TABLE learning_sequences_coursesection ALTER COLUMN usage_key TYPE VARCHAR(255) COLLATE "C";',
+                reverse_sql=migrations.RunSQL.noop, ),
+            migrations.RunSQL(
+                'ALTER TABLE learning_sequences_learningsequence ALTER COLUMN usage_key TYPE VARCHAR(255) COLLATE "C";',
+                reverse_sql=migrations.RunSQL.noop, ),
+        ]
 
-        # Custom code: Convert columns to utf8_bin because we want to allow
-        # case-sensitive comparisons for things like UsageKeys, CourseKeys, and
-        # slugs.
+    return [
         migrations.RunSQL(
             'ALTER TABLE learning_sequences_learningcontext MODIFY context_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
-            reverse_sql=migrations.RunSQL.noop,
-        ),
+            reverse_sql=migrations.RunSQL.noop, ),
         migrations.RunSQL(
             'ALTER TABLE learning_sequences_coursesection MODIFY usage_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
-            reverse_sql=migrations.RunSQL.noop,
-        ),
+            reverse_sql=migrations.RunSQL.noop, ),
         migrations.RunSQL(
             'ALTER TABLE learning_sequences_learningsequence MODIFY usage_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
-            reverse_sql=migrations.RunSQL.noop,
-        ),
+            reverse_sql=migrations.RunSQL.noop, ),
     ]
+
+
+class Migration(migrations.Migration):
+    initial = True
+    db_engine = connection.settings_dict['ENGINE']
+
+    dependencies = [
+    ]
+
+    operations = [
+                     migrations.CreateModel(
```

Comment:
If https://github.com/openedx/openedx-learning/pull/422 merges, would this be re-done in that style?

Reply:
I can certainly take care of that if desired.